### DISCUSSION TITLE
[Q&A] how to write external fluent-bit plugins

### DISCUSSION THREAD

**Discussion by TheDaz-1** [fluent/fluentd#4546] [Q&A]
### What is a problem?

I read through the developer documentation on how to add plugins to fluent-bit and couldn't find any details on how to create an external plugin that gets pulled in via the `[PLUGINS]` section of the configuration. I managed to create a plugin compiled into the fluent-bit binary but moving it into its own shared library and referencing it via the aforementioned configuration lets fluent-bit fail to resolve the plugin name.

This is the plugins registration section (mostly copied from the `out_stdout` example) :

```
/* Plugin registration */
struct flb_output_plugin out_test_plugin = {
    .name         = "test",
    .description  = "writes events to the test infrastructure",
    .cb_init      = cb_test_init,
    .cb_flush     = cb_test_flush,
    .cb_exit      = cb_test_exit,
    .flags        = 0,
    .workers      = 1,
    .event_type   = FLB_OUTPUT_LOGS,
    .config_map   = config_map
};
```

I wonder if the name of the registration map variable needs to be something more generic to allow for a generic loader to find it. Any idea ?

### Describe the configuration of Fluentd

`fluent-bit.conf` :
```
[SERVICE]
    ...
    plugins_file /my_dir/plugins.conf
    ...
[OUTPUT]
    Name test
    Match *
```

`plugins.conf` :
```
[PLUGINS]
    /my_dir/libTestOutputPlugin.so
```

### Describe the logs of Fluentd

`[error] [config] section 'test' tried to instance a plugin name that doesn't exist`

### Environment

```markdown
- fluent-bit version: 3.1.0
- Operating system: RHEL 9
```

**Comment by daipom** [comment]
This is Fluentd repository, not fluent-bit.
fluent-bit repository is https://github.com/fluent/fluent-bit.
